subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptionFactory.java

```diff
@@ -85,8 +85,9 @@ public void applyTo(StartParameter settings, Origin origin) {
     }
 
     public static class RecompileScriptsOption extends EnabledOnlyBooleanBuildOption<StartParameter> {
+        public static final String DEPRECATION_MESSAGE = "Support for --recompile-scripts was deprecated and is scheduled to be removed in Gradle 5.0.";
```

Comment:
Wondering if we could separate the message generation logic in `DeprecationLogger` from the console logging aspect. Then you could use the message generation here without having to hard-code the message in this class.